force-app/main/default/lwc/pearEventManager/pearEventManager.js

```diff
@@ -0,0 +1,69 @@
+import { LightningElement, api } from 'lwc';
+import { subscribe, unsubscribe, onError}  from 'lightning/empApi';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+
+export default class PearEventManager extends LightningElement {
+
+    subscription = {};
+     @api channel = '/event/AssetCancelInitiatedEvent';
+
+    // Initializes the component
+    connectedCallback() {
+        // Register error listener
+        this.registerErrorListener();
+        this.handleSubscribe();
+    }
+
+    disconnectedCallback() {
+        this.handleUnsubscribe();
+    }
+
+    // Handles subscribe button click
```

Comment:
What button?

Reply:
I think we can remove the unsubscribe function

Reply:
You need to clean up any event listeners and subscribers you create, but the comment refers to a button, and I don't think this component has a button? Or am I missing something?